OTCamera/helpers/name.py

```diff
@@ -87,22 +92,50 @@ def preview() -> str:
     return str(Path(filename).expanduser().resolve())
 
 
-def get_date_from_log_file(log_filepath: Path) -> dt:
-    regex = "[0-9]{4}-[0-9]{2}-[0-9]{2}_[0-9]{2}-[0-9]{2}-[0-9]{2}"
-    timestamp = re.findall(pattern=regex, string=log_filepath.stem)[-1]
-    date, time = timestamp.split("_")
-    year, month, day = date.split("-")
-    hour, minute, second = time.split("-")
-    dt_object = dt(
-        year=int(year),
-        month=int(month),
-        day=int(day),
-        hour=int(hour),
-        minute=int(minute),
-        second=int(second),
-    )
-    return dt_object
+def get_datetime_from_filename(filename: Union[str, Path]) -> dt:
+    """Retrieves the date and time from a filename
+
+    Searches for "_yyyy-mm-dd_hh-mm-ss".
+
+    Args:
+        filename (str): filename with expression
+
+    Returns:
+        dt: datetime object
+    """
+    if isinstance(filename, Path):
+        filename = filename.stem
+
+    regex = "_([0-9]{4,4}-[0-9]{2,2}-[0-9]{2,2}_[0-9]{2,2}-[0-9]{2,2}-[0-9]{2,2})"
+    match = re.search(regex, filename)
+    if not match:
+        return None
+
+    # Assume that there is only one timestamp in the file name
+    datetime_str = match.group(1)  # take group withtout underscore
+
+    try:
+        date_time = dt.strptime(datetime_str, "%Y-%m-%d_%H-%M-%S")
+    except ValueError:
+        return None
+
+    return date_time
+
+
+def get_fps_from_filename(filename: str) -> int:
```

Comment:
Shouldn't we refactor methods only used by tests to the test itself?

Reply:
Ah yes I added the method and its respective unit tests to the codebase in case we want to use it in the future.